### PR TITLE
Add model predict api

### PR DESCRIPTION
**Pull request by SeaOfOcean**
This pr is based on the change of https://github.com/intel-analytics/analytics-zoo/pull/62

### CODE REVIEW

File: pipeline/objectDetection/common/src/main/scala/com/intel/analytics/zoo/pipeline/common/model/Model.scala [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+      val miniBatch = localTransformer(partition)
+      miniBatch.flatMap(batch => {
+        val result = localModel.forward(batch.getInput()).toTensor[T]
+        val batchOut = if (result.dim() == 1) {
+          Array(result)
+        } else {
+          result.split(1)
+        }
+        batch.imageFeatures.zip(batchOut).map(x => {
+          x._1(ImageFeature.feature) = x._2
```
Comment (jason-dai) [inline pipeline/objectDetection/common/src/main/scala/com/intel/analytics/zoo/pipeline/common/model/Model.scala]: Change to ImageFeature.predit or ImageFeature.output

**Comment by jason-dai** [comment]
We can first merge the ImageFrame support to BigDL, and then add `predict(ImageFrame)` to AbstractModule and Predictor.

And instead of introducing new ImageFeatureMiniBatch and ImageFeatureToBatch, we can probably reuse the existing MiniBatch and SampleToMiniBatch as follows in `predict(ImageFrame)` :

```scala
    data.rdd.mapPartitions { partition =>
      val localModel = modelBroad.value()
      val localTransformer = broadcastToBatch.value.cloneTransformer()
      val localPostProcessor = broadcastPostProcess.value
      
      partition.grouped(size).flatMap { b =>
        val batch = localTransformer(b.iterator).next // need to move buffers out as class members
        val result = localModel.forward(batch.getInput()).toTensor[T]
        val batchOut = if (result.dim() == 1) {
          Array(result)
        } else {
          result.split(1)
        }
        b.zip(batchOut).map({
           ...
        })
      }
```

File: pipeline/objectDetection/common/src/main/scala/com/intel/analytics/zoo/pipeline/common/model/Model.scala [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+class Model[A <: Activity : ClassTag, B <: Activity : ClassTag, T: ClassTag]
+(model: AbstractModule[A, B, T])(
+  implicit ev: TensorNumeric[T]) {
+
+  SpatialShareConvolution.shareConvolution(model)
+  model.evaluate()
+
+  // get tag
+  val modelTag = "ssd_vgg16_300x300_pascal"
```
Comment (wzhongyuan) [inline pipeline/objectDetection/common/src/main/scala/com/intel/analytics/zoo/pipeline/common/model/Model.scala]: Supposed to get  from moduleName ?

File: pipeline/objectDetection/common/src/main/scala/com/intel/analytics/zoo/pipeline/common/model/Model.scala [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+  val versionTag = "0.3.0"
+  val defaultBatchPerPartition = 4
+
+  private def defaultToBatch(totalBatch: Int, partitionNum: Int) = SampleToMiniBatch[T](
+    batchSize = totalBatch,
+    partitionNum = Some(partitionNum))
+
+  private def defaultPostProcessor: (ImageFeature => ImageFeature) = null
+
+  private def getResources(imf: ImageFrame, partitionNum: Int) = modelTag match {
```
Comment (wzhongyuan) [inline pipeline/objectDetection/common/src/main/scala/com/intel/analytics/zoo/pipeline/common/model/Model.scala]: How about move it into a separate utility ?

**Comment by SeaOfOcean** [comment]
@jason-dai 

```
      partition.grouped(size).flatMap { b =>
        val batch = localTransformer(b.iterator).next // need to move buffers out as class members
        val result = localModel.forward(batch.getInput()).toTensor[T]
        val batchOut = if (result.dim() == 1) {
          Array(result)
        } else {
          result.split(1)
        }
        b.zip(batchOut).map({
           ...
        })
      }
```

In this way, the transformer will run twice?

**Pull request closed without merging**